vernac/vernacinterp.mli

```diff
@@ -20,8 +20,8 @@ val interp_qed_delayed_proof
   -> Vernacexpr.proof_end CAst.t
   -> Vernacstate.t
 
-(** [with_fail ~st f] runs [f ()] and expects it to fail, otherwise it fails. *)
-val with_fail : st:Vernacstate.t -> (unit -> 'a) -> unit
+(** [with_fail ~loc ~st f] runs [f ()] and expects it to fail, otherwise it fails. *)
+val with_fail : loc:Loc.t option -> st:Vernacstate.t -> (unit -> 'a) -> unit
```

Comment:
Why not a `?loc` instead?

Reply:
Because we don't want to forget to pass it